test/endToEnd/authorization/usersTest.js

```diff
@@ -184,6 +184,132 @@ function Factory (lab, data) {
   return records
 }
 
+function BuildFor (lab, records) {
+  return new TestBuilder(lab, records)
+}
+
+class TestBuilder {
+
+  constructor (lab, records) {
+    this.lab = lab
+    this.records = records
+  }
+
+  endpoint (endpointData) {
+    this.endpointData = endpointData
+    return this
+  }
+
+  server (serverInstance) {
+    this.serverInstance = serverInstance
+    return this
+  }
+
+  test (description) {
+    const test = new CustomTest(this)
+    test.test(description)
+    return test
+  }
+
+}
+
+function getValueFromPath (data, key) {
+  const keys = key.split('.')
+  const localKey = keys.shift()
+  const localValue = data[localKey]
+
+  if (keys.length === 0) {
+    return localValue
+  }
+
+  if (_.isUndefined(localValue) || _.isNull(localValue)) {
+    return undefined
+  }
+
+  return getValueFromPath(localValue, keys.join('.'))
+}
+
+function interpolate (value, data) {
+  function interpolator (value) {
+    return interpolate(value, data)
+  }
+
+  if (_.isArray(value)) {
+    return _.map(value, interpolator)
+  }
+
+  if (_.isObject(value)) {
+    return _.mapValues(value, interpolator)
+  }
+
+  if (!_.isString(value)) {
```

Comment:
I don't think we'll use String objects, but it's better to fix this and have a robust solution.
I'll switch order.